x-pack/plugins/fleet/server/services/epm/elasticsearch/template/template.test.ts

```diff
@@ -23,27 +23,29 @@ expect.addSnapshotSerializer({
 });
 
 test('get template', () => {
```

Comment:
it would be great to have a unit test for `installTemplate` since that's the method that generates the index pattern and priority based on data stream information. I don't see any tests here that actually test on the `dataset_is_prefix` property

Reply:
Good point. Is there an established way how to mock calls to elasticsearch in unit tests?

I've added unit tests for `generateTemplateIndexPattern()` and `getTemplatePriority()` as a start in https://github.com/elastic/kibana/pull/89870/commits/8b58f46130659db90951620311752e23d3057ce3

Reply:
it looks like Kibana core does export some ES mocks, here's an example: https://github.com/elastic/kibana/blob/master/x-pack/plugins/fleet/server/services/epm/packages/_install_package.test.ts#L38

I see that `installTemplate` passes `callCluster` to other methods, so I guess you might have to mock ES return values in order of whenever `callCluster` is called further down the chain. the mocked ES client returns a `jest.fn()` so mocking the responses could be done like this:

```
const mockCallCluster = elasticsearchServiceMock.createLegacyScopedClusterClient().callAsCurrentUser
  .mockReturnValueOnce({ success: true })
  .mockReturnValueOnce({ success: true })
  .mockReturnValueOnce(// change mocked response whenever the response matters)
  ...
```

Reply:
Thanks for the pointer, I've added a unit test for `installTemplate()`.